Issue 8

Khyzad opened the pull request:
Removed incrementPid() in sequentialMint()

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/hawaiistatedigitalarchives/pid-webservice/24)

<!-- Reviewable:end -->

Khyzad committed:
Remove incrementPid in sequentialMint

additional changes:
- corrected javadocs in test cases
- throw negative NotEnoughPermutationException when amount is negative
- expect NotEnoughPermutationException when amount is negative in test

Khyzad committed:
Merge branch 'Springboot-feature' into issue-8

dagnir commented:
Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

_[Minter/src/main/java/com/hida/model/IdGenerator.java, line 85 [r1]](https://reviewable.io:443/reviews/hawaiistatedigitalarchives/pid-webservice/24#-KLyYZSvuDp4-t1dsMpr:-KLyYZSvuDp4-t1dsMps:1454812209) ([raw file](https://github.com/hawaiistatedigitalarchives/pid-webservice/blob/5215c385a7490748636abe563467a1b7b9a75320/Minter/src/main/java/com/hida/model/IdGenerator.java#L85)):_

> ``` Java
>     public Set<Pid> sequentialMint(long amount) {
>         // checks to see if its possible to produce or add requested amount of
>         if (MaxPermutation < amount || amount < 0) {
> ```

Let's throw `IllegalArgumentException` instead if amount is < 0.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/hawaiistatedigitalarchives/pid-webservice/24)_

<!-- Sent from Reviewable.io -->

Khyzad committed:
Throw IllegalArgumentException when negative

additional changes:
- correct negative amount test in *GeneratorTest
- correct javadocs in *GeneratorTest

dagnir commented:
Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/hawaiistatedigitalarchives/pid-webservice/24)_

<!-- Sent from Reviewable.io -->

dagnir commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

_[Minter/src/main/java/com/hida/model/IdGenerator.java, line 114 [r2]](https://reviewable.io:443/reviews/hawaiistatedigitalarchives/pid-webservice/24#-KM0g-Ouw073oBPyFeua:-KM0g-OvsKmAw4YjdYPU:-1627818959) ([raw file](https://github.com/hawaiistatedigitalarchives/pid-webservice/blob/ecd07d4ff70a6a146506e93151782fd4c5f89215/Minter/src/main/java/com/hida/model/IdGenerator.java#L114)):_

> ``` Java
>      */
>     public Set<Pid> sequentialMint(long amount, long startingValue) {
>         if (MaxPermutation < amount) {
> ```

Sorry I missed this one. Also, looks like `sequentialMint(long amount)` can be written in terms of this one by using a `startingValue` of 0

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/hawaiistatedigitalarchives/pid-webservice/24)_

<!-- Sent from Reviewable.io -->

Khyzad committed:
Throw IllegalArgumentException when negative

Khyzad committed:
Throw exception if startingValue exceeds maximum

dagnir commented:
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

_[Minter/src/main/java/com/hida/model/IdGenerator.java, line 83 [r3]](https://reviewable.io:443/reviews/hawaiistatedigitalarchives/pid-webservice/24#-KMGt8UQlJY7_QRRap4X:-KMGt8UQlJY7_QRRap4Y:1253686479) ([raw file](https://github.com/hawaiistatedigitalarchives/pid-webservice/blob/b763a753e5b908adf5b7b084b39fcac69b9cb342/Minter/src/main/java/com/hida/model/IdGenerator.java#L83)):_

> ``` Java
>      * @return A set of Pids
>      */
>     public Set<Pid> sequentialMint(long amount) {
> ```

This can be rewritten as

``` java
return sequentialMint(amount, 0);
```

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/hawaiistatedigitalarchives/pid-webservice/24)_

<!-- Sent from Reviewable.io -->

Khyzad committed:
Rewrite sequentialMint(long)

Logically, the content of sequentialMint(amount) can be written as
return sequentialMint(amount, 0);

dagnir commented:
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/hawaiistatedigitalarchives/pid-webservice/24)_

<!-- Sent from Reviewable.io -->

dagnir commented:
@Khyzad, please fix the merge conflicts when you're able
